test/parallel/test-cli-node-cli-manpage-options.mjs

```diff
@@ -0,0 +1,165 @@
+import '../common/index.mjs';
+import assert from 'assert';
+import { createReadStream, readFileSync } from 'node:fs';
+import { createInterface } from 'node:readline';
+import { resolve, join } from 'node:path';
+
+// This test checks that all the CLI flags defined in the public CLI documentation (doc/api/cli.md)
+// are also documented in the manpage file (doc/node.1)
+// Note: the opposite (that all variables in doc/node.1 are documented in the CLI documentation)
+//       is covered in the test-cli-node-options-docs.js file
+
+const rootDir = resolve(import.meta.dirname, '..', '..');
+
+const cliMdPath = join(rootDir, 'doc', 'api', 'cli.md');
+const cliMdContentsStream = createReadStream(cliMdPath);
+
+const manPagePath = join(rootDir, 'doc', 'node.1');
+const manPageContents = readFileSync(manPagePath, { encoding: 'utf8' });
+
+// TODO(dario-piotrowicz): add the missing flags to the node.1 and remove this set
+const knownFlagsMissingFromManPage = new Set([
+  'build-snapshot',
+  'build-snapshot-config',
+  'disable-sigusr1',
+  'disable-warning',
+  'dns-result-order',
+  'enable-network-family-autoselection',
+  'env-file-if-exists',
+  'env-file',
+  'experimental-network-inspection',
+  'experimental-print-required-tla',
+  'experimental-require-module',
+  'experimental-sea-config',
+  'experimental-worker-inspection',
+  'expose-gc',
+  'force-node-api-uncaught-exceptions-policy',
+  'import',
+  'network-family-autoselection-attempt-timeout',
+  'no-async-context-frame',
+  'no-experimental-detect-module',
+  'no-experimental-global-navigator',
+  'no-experimental-require-module',
+  'no-network-family-autoselection',
+  'openssl-legacy-provider',
+  'openssl-shared-config',
+  'report-dir',
+  'report-directory',
+  'report-exclude-env',
+  'report-exclude-network',
+  'run',
+  'snapshot-blob',
+  'trace-env',
+  'trace-env-js-stack',
+  'trace-env-native-stack',
+  'trace-require-module',
+  'use-system-ca',
+  'watch-preserve-output',
+]);
+
+const optionsEncountered = { dash: 0, dashDash: 0, named: 0 };
+let insideOptionsSection = false;
+
+const rl = createInterface({
+  input: cliMdContentsStream,
+});
+
+const isOptionLineRegex = /^###( `[^`]*`(,)?)*$/;
```

Comment:
Nit:
```suggestion
const isOptionLineRegex = /^###(?: `[^`]*`,?)*$/;
```
Ditto.

Reply:
oops, good spot! thanks, I think I left the `(` `)` from an earlier iteration of the regex :pray: